Tasks/IISWebAppManagementOnMachineGroup/task.json

```diff
@@ -23,59 +23,101 @@
 	"instanceNameFormat": "Manage IIS Web App: $(WebsiteName)",
 	"groups": [
 		{
-			"name": "website",
-			"displayName": "Website",
-			"isExpanded": true
+			"name": "Website",
+			"displayName": "IIS Website",
+			"isExpanded": true,
+			"visibleRule": "ActionIISWebsite = CreateOrUpdateWebsite"
 		},
 		{
-			"name": "applicationPool",
-			"displayName": "Application Pool",
-			"isExpanded": true
+			"name": "ApplicationPoolForWebsite",
+			"displayName": "IIS application pool",
+			"isExpanded": true,
+			"visibleRule": "IISDeploymentType = IISWebsite && ActionIISWebsite = CreateOrUpdateWebsite && CreateOrUpdateAppPoolForWebsite = true"
+		},
+		{
+			"name": "ApplicationPool",
+			"displayName": "IIS application pool",
+			"isExpanded": true,
+			"visibleRule": "ActionIISApplicationPool = CreateOrUpdateAppPool"
 		},
 		{
-			"name": "advanced",
+			"name": "ApplicationPoolForApplication",
+			"displayName": "IIS application pool",
+			"isExpanded": true,
+			"visibleRule": "IISDeploymentType = IISWebApplication && CreateOrUpdateAppPoolForApplication = true"
+		},
+		{
+			"name": "Advanced",
 			"displayName": "Advanced",
-			"isExpanded": false
+			"isExpanded": true
 		}
 	],
 	"inputs": [
 		{
-			"name": "CreateWebsite",
-			"type": "boolean",
-			"label": "Create or Update website",
+			"name": "IISDeploymentType",
+			"type": "pickList",
+			"label": "IIS Deployment Type",
 			"required": false,
-			"groupName": "website",
-			"defaultValue": "false",
-			"helpMarkDown": "Select the option to create a website or to update an existing website."
+			"options": {
+				"IISWebsite": "IIS Website",
+				"IISWebApplication": "IIS Web Application",
+				"IISVirtualDirectory": "IIS Virtual Directory",
+				"IISApplicationPool": "IIS Application Pool"
+			},
+			"helpMarkDown": ""
+		},
+		{
+			"name": "ActionIISWebsite",
+			"type": "pickList",
+			"label": "Action",
+			"required": true,
+			"visibleRule": "IISDeploymentType = IISWebsite",
+			"options": {
+				"CreateOrUpdateWebsite": "Create Or Update",
+				"StartWebsite": "Start website",
+				"StopWebsite": "Stop website"
+			},
+			"helpMarkDown": ""
+		},
+		{
+			"name": "ActionIISApplicationPool",
+			"type": "pickList",
+			"label": "Action",
+			"required": true,
+			"visibleRule": "IISDeploymentType = IISApplicationPool",
+			"options": {
+				"CreateOrUpdateAppPool": "Create Or Update",
+				"StartAppPool": "Start App pool",
```

Comment:
I assume the start/stop of App pool is not available either. Bring it in json once the backend is available.